addons/knowledge/static/src/components/permission_panel/permission_panel.js

```diff
@@ -0,0 +1,334 @@
+            $select.val(oldPermission);
+            this.loadPanel();
+        };
+        const loseAccessMessage = _t('Are you sure you want to set the internal permission to "none" ? If you do, you will no longer have access to the article.');
+        this._showConfirmDialog(willLoseAccess ? loseAccessMessage : restrictMessage, confirm, discard);
+    }
+
+    /**
+     * Callback function called when the permission of a user changes.
+     * @param {Event} event
+     * @param {Proxy} member
+     */
+    _onChangeMemberPermission (event, member) {
+        const index = this.state.members.findIndex(current => {
+            return current.id === member.id;
+        });
+        if (index < 0) {
+            return;
+        }
+        const $select = $(event.target);
+        const newPermission = $select.val();
+        const oldPermission = member.permission;
+        const willLoseAccess = this.isLoggedUser(member) && newPermission === 'none';
+        const willRestrict = this.state.based_on && permissionLevel[newPermission] < permissionLevel[oldPermission];
+        const willLoseWrite = this.isLoggedUser(member) && newPermission !== 'write' && oldPermission === 'write';
+        const confirm = () => {
+            this.rpc({
+                route: '/knowledge/article/set_member_permission',
+                params: {
+                    article_id: this.props.article_id,
+                    permission: newPermission,
+                    member_id: member.based_on ? false : member.id,
+                    inherited_member_id: member.based_on ? member.id: false,
+                }
+            }).then(res => {
+                const reloadArticleId = willLoseWrite && !willLoseAccess ? this.props.article_id : false;
+                if (this._onChangedPermission(res, willLoseAccess||willLoseWrite, reloadArticleId)) {
+                    this.loadPanel();
+                }
+            });
+        };
+
+        if (!willLoseAccess && !willRestrict && !willLoseWrite) {
+            confirm();
+            return;
+        }
+
+        const discard = () => {
+            $select.val(this.state.members[index].permission);
+            this.loadPanel();
+        };
+        const loseAccessMessage = _t('Are you sure you want to set your permission to "none"? If you do, you will no longer have access to the article.');
+        const message = willLoseAccess ? loseAccessMessage : willLoseWrite ? loseWriteMessage : loseAccessMessage;
+        this._showConfirmDialog(message, confirm, discard);
+    }
+
+    /**
+     * @param {Event} event
+     * @param {integer} id - article id
+     */
+    _onOpen (event, id) {
+        event.preventDefault();
+        this.openArticle(id);
+    }
+
+    /**
+     * Callback function called when a member is removed.
+     * @param {Event} event
+     * @param {Proxy} member
+     */
+    _onRemoveMember (event, member) {
+        const index = this.state.members.findIndex(current => {
```

Comment:
you can simply use `indexOf`: 
```js
const index = this.state.members.indexOf(member);
```
(assuming that member is the same object).
Also, in that case, and since you don't really need the index anyway, you can just do:
```js
if (!this.state.members.includes(member) {
  return;
```
```

Reply:
@fbarbe00